src/kafka.go

```diff
@@ -185,9 +187,24 @@ func coreCollection(kafkaIntegration *integration.Integration) {
 			os.Exit(2)
 		}
 
-		topics, err := topic.GetTopics(clusterClient)
-		if err != nil {
-			log.Error("Failed to get a list of topics. Continuing with broker collection: %s", err)
+		var topics []string
+		if args.GlobalArgs.TopicSource == discoverZookeeper && args.GlobalArgs.AutodiscoverStrategy == discoverZookeeper {
```

Comment:
This implicit requirement might hide some misconfigurations, I'd prefer to error out if `args.GlobalArgs.TopicSource == discoverZookeeper` but `args.GlobalArgs.AutodiscoverStrategy` is not `discoverZookeeper`